mesonbuild/mtest.py

```diff
@@ -1912,6 +1915,9 @@ async def run_test(test: SingleTestRunner) -> None:
                     return
                 res = await test.run(self)
                 self.process_test_result(res)
+                maxfail = self.options.maxfail
+                if maxfail and self.fail_count >= maxfail and res.res.is_bad():
```

Comment:
I could have foggy brain in the morning, but isn't that off-by-one? If maxfail=1 and the first test fails, fail_count=0, so it won't stop until a 2nd failure, right?

Reply:
Forget me, counter is incremented in process_test_result() called just before.

Reply:
It's incremented in `process_test_result()`, that's why it has to wait until right after we do that.